### PR TITLE
RFC for drop schema design

### PR DESCRIPTION
**Pull request by jfunston**

### CODE REVIEW

File: docs/RFC/drop_schema.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+1. User request: truncateDataAndDropSchema(name)
+2. Client: the client library translates the name to the schema ID
+3. Client: Request sent to CPO
+4. CPO: drops schema locally (will no longer be returned by getSchema requests)
+5. CPO: sends truncateDataAndDropSchema to all partitions
+6. Partition: remove schema locally (no longer accepts requests for that schema)
+7. Partition: responds to CPO
+8. CPO: after receiving responses from all partitions, responds to client. At this point the user may recreate a schema with the same name
+9. Partition (async): scan and truncate all records and all MVCC versions matching the schema. No persistence or tombstones are needed.
+
```
Comment (iafuture) [inline docs/RFC/drop_schema.md:41]: WAL cleanup; hot standby?

**Review by iafuture — APPROVED** [review]

File: docs/RFC/drop_schema.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+truncateDataAndDropSchema truncates all data and all schema references for a given schema name (i.e. it deletes all versions of the schema).
+
+Operations:
+1. User request: truncateDataAndDropSchema(name)
+2. Client: the client library translates the name to the schema ID
+3. Client: Request sent to CPO
+4. CPO: drops schema locally (will no longer be returned by getSchema requests)
+5. CPO: sends truncateDataAndDropSchema to all partitions
+6. Partition: remove schema locally (no longer accepts requests for that schema)
```
Comment (jerryhfeng) [inline docs/RFC/drop_schema.md:31]: At partition, schema exists only in memory. So let's make it clear that remove is for in memory only.

File: docs/RFC/drop_schema.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+7. Partition: responds to CPO
+8. CPO: after receiving responses from all partitions, responds to client. At this point the user may recreate a schema with the same name
```
Comment (jerryhfeng) [inline docs/RFC/drop_schema.md:33]: Do we have any plan to gracefully handle when one or a few partition didn't reply promptly due to various reason?

File: docs/RFC/drop_schema.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+9. Partition (async): scan and truncate all records and all MVCC versions matching the schema. No persistence or tombstones are needed.
```
Comment (jerryhfeng) [inline docs/RFC/drop_schema.md:34]: This is for local in Memory(DRAM) data, right? If we will put data into local Plog based NVDIMM space, we can't really reclaim those space till GC. Let's make that clear.

**Review by jerryhfeng — APPROVED** [review]

**Review by iafuture — APPROVED** [review]

**Pull request merged**